force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -318,17 +405,29 @@ public with sharing class ClassesObjectsMethods {
      */
 
     public static List<String> personDetails() {
-        List<String> results = new List<String>();
+        
         // Create a new instance of Person class
         // Name the person 'John Doe' and set the age to 28
+        String expectedPersonName = 'John Doe';
+        Integer expectedPersonAge = 28;
+
+        ClassesObjectsMethods.Person myNewPerson = new ClassesObjectsMethods.Person(expectedPersonName, expectedPersonAge);
+
+        // Create a list of books
+        List<String> personList = new List<String>();
 
         // Add the details of the person using the introduceYourself method
+        String intro = myNewPerson.introduceYourself();
+        personList.add(intro);
 
         // Get the name of the person using the getter method and add it to the list
+        String name = myNewPerson.getName();
+        personList.add(name);
 
         // Get the age of the person using the getter method and add it to the list
-        
-        return null; // Replace null with the variable you used to store the result
+        Integer age = myNewPerson.getAge();
+        personList.add(String.valueOf(age));
+        return personList; 
```

Comment:
Again way more code that you need. There is no value in creating variables for no reason than to pass them to the constructor.     Person person = new Person('John Doe', 28);